File: python_programming_basics/while_loops/l_08_graduation.py
student_name = input()
current_class = 1
bad_grades = 0
average_grade = 0
failed = False

while current_class <= 12:
    year_grade = float(input())

    if year_grade < 4:
        bad_grades += 1
        if bad_grades > 1:
            failed = True
            break
        continue

    current_class += 1
    average_grade += year_grade / 12

if failed:
    print(f"{student_name} has been excluded at {current_class} grade")

else:
    print(f"{student_name} graduated. Average grade: {average_grade:.2f}")

File: JavaScript Basics/05. While Loop/l-08-Graduation.js
function calculateStudentPerformance(input) {
    let name = input[0];
    let index = 1;
    let grade = 1;
    let sumGrades = 0;
    let failCount = 0;

    while (grade <= 12) {
        let annualGrade = parseFloat(input[index]);
        index++;

        if (annualGrade < 4.0) {
            failCount++;
            if (failCount > 1) {
                console.log(`${name} has been excluded at ${grade} grade`);
                return;
            }
        } else {
            sumGrades += annualGrade;
            grade++;
        }
    }

    let averageGrade = sumGrades / 12;
    console.log(`${name} graduated. Average grade: ${averageGrade.toFixed(2)}`);
}